forecasting/src/autogluon/forecasting/predictor.py

```diff
@@ -0,0 +1,420 @@
+                **kwargs,
+            )
+            logger.info("============ fit arguments ============")
+            logger.info("fit() called with arguments:")
+            logger.info(f"{pprint.pformat(fit_args)}")
+            logger.info("=======================================")
+
+        # Inform the user extra columns in dataset will not be used.
+        extra_columns = [c for c in train_data.columns.copy() if c != target_column]
+        if len(extra_columns) > 0:
+            logger.warning(f"Provided columns {extra_columns} will not be used.")
+
+        if val_data is None:
+            logger.warning(
+                f"Validation data is None, will hold the last prediction_length {prediction_length} "
+                f"time steps out to use as validation set.",
+            )
+            val_data = train_data
+            train_data = train_data.slice_by_timestep(slice(None, -prediction_length))
+
+        quantile_levels = kwargs.get(
+            "quantile_levels",
+            kwargs.get("quantiles", [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9]),
+        )
+        logger.info(f"All models will be trained for quantiles {quantile_levels}.")
+
+        scheduler_options = self._get_scheduler_options(
+            hyperparameter_tune_kwargs, time_limit=time_limit
+        )
+        time_left = (
+            None if time_limit is None else time_limit - (time.time() - start_time)
+        )
+        self._learner.fit(
+            train_data=train_data,
+            prediction_length=prediction_length,
+            val_data=val_data,
+            scheduler_options=scheduler_options,
+            hyperparameters=hyperparameters,
+            hyperparameter_tune=all(scheduler_options),
+            quantile_levels=quantile_levels,
+            time_limit=time_left,
+        )
+
+        self.save()
+        return self
+
+    # TODO: to be changed after ray tune integration
+    def _get_scheduler_options(
+        self,
+        hyperparameter_tune_kwargs: Optional[Union[str, Dict]],
+        time_limit: Optional[int] = None,
+    ) -> Tuple[Optional[Type], Optional[Dict[str, Any]]]:
+        """Validation logic for `hyperparameter_tune_kwargs`. Returns True if `hyperparameter_tune_kwargs` is None or
+        can construct a valid scheduler. Returns False if hyperparameter_tune_kwargs results in an invalid scheduler.
+        """
+        if hyperparameter_tune_kwargs is None:
+            return None, None
+
+        num_trials: Optional[int] = None
+        if isinstance(hyperparameter_tune_kwargs, dict):
+            num_trials = hyperparameter_tune_kwargs.get("num_trials")
+            if time_limit is None and num_trials is None:
+                logger.warning(
+                    "None of time_limit and num_trials are set, defaulting to num_trials=2",
+                )
+                num_trials = 2
+            else:
+                num_trials = hyperparameter_tune_kwargs.get("num_trials", 9999)
```

Comment:
Note: We may want to reign this in eventually. Running out of disk space becomes a real problem when you save 9999 models over the course of a few hours. This applies to the other modules as well.